tools/_venv_common.py

```diff
@@ -8,11 +8,91 @@
 import time
 import subprocess
 import sys
+import re
+
+VERSION_PATTERN = re.compile(r'^(\d+)\.(\d+).*$')
+
+
+class PythonExecutableNotFoundError(Exception):
+    pass
+
+
+def find_python_executable(python_major):
+    # type: (int) -> str
+    """
+    Find the relevant python executable that is of the given python major version.
+    Will test, in decreasing priority order:
+    * the current Python interpreter
+    * 'pythonX' executable in PATH (with X the given major version) if available
+    * 'python' executable in PATH if available
+    * Windows Python launcher 'py' executable in PATH if available
+    Incompatible python versions for Certbot will be evicted (eg. Python < 3.5 on Windows)
+    :param int python_major: the Python major version to target (2 or 3)
+    :rtype: str
+    :return: the relevant python executable path
+    :raise: RuntimeError if no relevant python executable path could be found
```

Comment:
It's a little odd, but the `:` after `raise` here should be moved after `RuntimeError`. See http://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists for examples.

Reply:
Corrected